### PR TITLE
Added missing include dirs + fixes

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Fixes building the package when using `azure-iot-sdk-c` as package by adding the missing `PLATFORM_INC_DIR` CMake variable.
* Cleanup some unused variables

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by adteran — COMMENTED** [review]
Please separate the changes required to address building from cleanup.

**Review by adteran — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Comment by MariusNi** [comment]
Merge from latest (as of 7/12) to have all CI pass. This code is not in sync with main as the PR changelist declares.

File: src/agents/pnp/CMakeLists.txt [hunk 1]
```diff
@@ -80,7 +80,8 @@ else()
 
     target_include_directories(${target_name} PUBLIC
         ${SHARED_UTIL_INC_FOLDER}
-        ${MACRO_UTILS_INC_FOLDERP})
+        ${MACRO_UTILS_INC_FOLDERP}
+        ${PLATFORM_INC_DIR})
```
Comment (MariusNi) [inline src/agents/pnp/CMakeLists.txt:84]: > ${PLATFORM_INC_DIR}) [](http://example.com/codeflow?start=7&length=21)

Is this really required here for Percept? It's not needed on Ubuntu and Debian.

**Review by MariusNi — APPROVED** [review]
:shipit:

File: src/agents/pnp/CMakeLists.txt (hunk 1 above)
Comment (AhmedBM) [reply src/agents/pnp/CMakeLists.txt:84]: its needed if we are using `azure-iot-sdk` as package, regardless of the distro

**Pull request merged**